### PR TITLE
update mwaa

### PR DESCRIPTION
**Pull request by Tian-2017**
This MWAA Terraform code works great now in the dev account. I have submitted the version I used for the final test.

I am not using the module for bucket creation because it causes a failure in log group creation in CloudWatch. This results in us being unable to see the logs in the Airflow UI, leading to the failure of MWAA creation if logs are turned on. Therefore, I have to tailor the KMS policy.

I greatly appreciate Wayne and Matt's help in debugging the networking issue and providing hints for others too!

### CODE REVIEW

File: terraform/core/47-mwaa.tf [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17",
+    Statement = [
+      {
+        Action = "sts:AssumeRole",
+        Effect = "Allow",
+        Principal = {
+          Service = [
+            "airflow-env.amazonaws.com",
+            "airflow-env.amazonaws.com"
```
Comment (LBHMKeyworth) [inline terraform/core/47-mwaa.tf:13]: Not sure this needs to be duplicated.

File: terraform/core/46-mwaa-bucket-kms.tf [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        Resource = "*"
+      },
+      {
+        Sid    = ""
+        Effect = "Allow"
+        Principal = {
+          AWS = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:root"
+        }
+        Action   = "kms:*"
+        Resource = "*"
```
Comment (LBHMKeyworth) [inline terraform/core/46-mwaa-bucket-kms.tf:29]: This allows all actions on the key

File: terraform/core/46-mwaa-bucket-kms.tf [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+          AWS = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:root"
+        }
+        Action = [
+          "kms:RetireGrant",
+          "kms:ReEncrypt*",
+          "kms:GenerateDataKey*",
+          "kms:Encrypt",
+          "kms:DescribeKey",
+          "kms:Decrypt",
+          "kms:CreateGrant"
```
Comment (LBHMKeyworth) [inline terraform/core/46-mwaa-bucket-kms.tf:44]: This seems to be giving more permissions to the same principal as the section above which give all permissions

**Review by LBHMKeyworth — CHANGES_REQUESTED** [review]
A couple of little things to look at.

File: terraform/core/47-mwaa.tf (hunk 1 above)
Comment (Tian-2017) [reply terraform/core/47-mwaa.tf]: Aha, my fault. I did some formatting and renaming, not sure how I changed this.

File: terraform/core/46-mwaa-bucket-kms.tf (hunk 2 above)
Comment (Tian-2017) [reply terraform/core/46-mwaa-bucket-kms.tf:29]: I only allow Logs to use the KMS key. The other two sections are used on all the keys in our AWS account. Could we keep it as same as other now. What do you think?
You can check all the keys for example in the **DataPlatform-Pre-Production** account via below URL?
https://eu-west-2.console.aws.amazon.com/kms/home?region=eu-west-2#/kms/keys

File: terraform/core/46-mwaa-bucket-kms.tf [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+# Create an alias for the KMS key for better readability in the console
+resource "aws_kms_alias" "mwaa_key_alias" {
+  name          = "alias/mwaa-key"
+  target_key_id = aws_kms_key.mwaa_key.key_id
+}
+
+# Create the S3 bucket using the KMS key
+resource "aws_s3_bucket" "mwaa_bucket" {
+  bucket = "dataplatform-${local.short_identifier_prefix}mwaa-bucket"
```
Comment (timburke-hackit) [inline terraform/core/46-mwaa-bucket-kms.tf:67]: this is equivalent and how the other buckets are named
```suggestion
  bucket = "${local.identifier_prefix}-mwaa-bucket"
```

Comment (Tian-2017) [reply terraform/core/46-mwaa-bucket-kms.tf]: @timburke-hackit Many thanks!

File: terraform/core/47-mwaa.tf [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+
+resource "aws_s3_bucket_object" "requirements_placeholder" {
+  bucket  = aws_s3_bucket.mwaa_bucket.bucket
+  key     = "requirements/.placeholder"
+  content = ""
+}
+
+resource "aws_mwaa_environment" "mwaa" {
+  # count              = local.is_live_environment && !local.is_production_environment ? 1 : 0
+  name               = "dataplatform-${local.short_identifier_prefix}mwaa-environment"
```
Comment (Tian-2017) [inline terraform/core/47-mwaa.tf:144]: Hey Tim, same as the bucket name. Is it better to use `${local.identifier_prefix}-mwaa-environment` here? Thanks @timburke-hackit

Comment (timburke-hackit) [reply terraform/core/47-mwaa.tf:144]: yeah, probably - it's not a big deal but maybe makes it easier to have all the mwaa resources named consistently?

**Comment by timburke-hackit** [comment]
not sure if you're planning to test this and revert(?), but as is you're creating all the resources in prod and dev if someone were to deploy main.

**Comment by Tian-2017** [comment]
> not sure if you're planning to test this and revert(?), but as is you're creating all the resources in prod and dev if someone were to deploy main.

I've terraformed all resources in our Dev account, the airflow works well there now.
I am thinking to deploy to staging and prod to make sure it works in both. Then we can add one line (the line of count) to destroy the mwaa in prod. @timburke-hackit

**Comment by timburke-hackit** [comment]
@Tian-2017 cool, yeah that's what I meant - whether you were planning to test it in prod too, which makes sense - all good 👍

**Review by LBHMKeyworth — APPROVED** [review]

**Review by LBH-wgreeff — APPROVED** [review]
LGTM

**Pull request merged**